server/src/main/java/com/google/musicanalysis/site/AnalysisServlet.java

```diff
@@ -0,0 +1,157 @@
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/api/analysis")
+public class AnalysisServlet extends HttpServlet {
+  @Override
+  protected void doGet(HttpServletRequest req, HttpServletResponse res)
+      throws ServletException, IOException {
+
+    res.getWriter().write("<h1>Song Analysis</h1>\n");
+
+    String trackName = "baby";
+    String artistName = "justin bieber";
+
+    res.getWriter().write("<h3>Lyrics</h3>");
+    String trimmedLyrics = getLyrics(trackName, artistName);
+    res.getWriter().println(trimmedLyrics);
+
+    res.getWriter().write("<h3>Perspective</h3>");
+    HashMap<String, String> perspectiveMap = analyzeWithPerspective(trimmedLyrics);
+    // Print on the screen (won't make it in final cut)
+    Set set = perspectiveMap.entrySet();
+    Iterator mapIterator = set.iterator();
+    while (mapIterator.hasNext()) {
+      Map.Entry entry = (Map.Entry) mapIterator.next();
+      res.getWriter().println(entry.getKey() + ": " + entry.getValue());
+    }
+
+    res.getWriter().write("<h3>NLP</h3>");
+    MagnitudeAndScore nlpObject = analyzeWithNLP(trimmedLyrics);
+    res.getWriter().println("Magnitude: " + nlpObject.magnitude + "Score: " + nlpObject.score);
+  }
+
+  private class MagnitudeAndScore {
+    Double magnitude;
+    Double score;
+
+    public MagnitudeAndScore(Double magnitude, Double score) {
+      this.magnitude = magnitude;
+      this.score = score;
+    }
+  }
+
+  private MagnitudeAndScore analyzeWithNLP(String text) throws IOException {
+
+    // Calling the API
+    String response = new SentimentRequest(text).getResponse();
+
+    // Extracting the sentiment
+    JsonElement jElement = JsonParser.parseString(response);
+    JsonObject jObject = jElement.getAsJsonObject();
+    JsonObject sentimentObject = jObject.getAsJsonObject("documentSentiment");
+
+    MagnitudeAndScore nlpResults =
+        new MagnitudeAndScore(
+            Double.valueOf(sentimentObject.get("magnitude").toString()),
+            Double.valueOf(sentimentObject.get("score").toString()));
+
+    return nlpResults;
+  }
+
+  private HashMap<String, String> analyzeWithPerspective(String text) throws IOException {
+
+    ArrayList<String> attributes =
+        new ArrayList<>(
+            Arrays.asList(
+                "TOXICITY",
+                "IDENTITY_ATTACK",
+                "INSULT",
+                "PROFANITY",
+                "THREAT",
+                "SEXUALLY_EXPLICIT",
+                "FLIRTATION"));
+
+    // Calling the API
+    PerspectiveRequest perspectiveRequest = new PerspectiveRequest(text, attributes);
+    String response = perspectiveRequest.getResponse();
+
+    // Extracting the results
+    JsonElement jElement = JsonParser.parseString(response);
+    JsonObject jObject = jElement.getAsJsonObject();
+    JsonObject attributeObject = jObject.getAsJsonObject("attributeScores");
```

Comment:
I see this section that "extracts the results" is repeated  like in lines 65-68 above except for the input parameter `"attributeScores"`, is it possible to create a helper fn that reduces duplicate code?

Reply:
I see what you're saying here...the only reason why I'm hesitant is that sometimes I need to getAsJsonObject(), and sometimes I need to getAsJsonArray().

Reply:
That would require me:
1. Having a JsonObject helper function and a Json Array helper
or 
2. Having some sort of extra value I pass in that determines wheher to get as Array or get as Object. 

I'm not opposed to it but I'm not sure that I would use the helper more than twice. However I am biased towards my own code lol.

Reply:
okay, got it! Then my request doesn't seem to helpful. Thank you taking a step back to think about your code.